app/code/Magento/CatalogCustomerGraphQl/Model/Resolver/PriceTiers.php

```diff
@@ -120,42 +120,89 @@ function () use ($productId, $context) {
 
                 $productPrice = $this->tiers->getProductRegularPrice($productId) ?? 0.0;
                 $tierPrices = $this->tiers->getProductTierPrices($productId) ?? [];
-
-                return $this->formatProductTierPrices($tierPrices, $productPrice, $store);
+                return $this->filterAndFormatProductTierPrices($tierPrices, $productPrice, $store);
             }
         );
     }
 
     /**
-     * Format tier prices for output
+     * Filter and format tier prices for output
      *
      * @param ProductTierPriceInterface[] $tierPrices
      * @param float $productPrice
      * @param StoreInterface $store
      * @return array
      */
-    private function formatProductTierPrices(array $tierPrices, float $productPrice, StoreInterface $store): array
-    {
+    private function filterAndFormatProductTierPrices(
+        array $tierPrices,
+        float $productPrice,
+        StoreInterface $store
+    ): array {
         $tiers = [];
+        $qtyCache = [];
+
+        foreach ($tierPrices as $key => $tierPrice) {
+            $this->formatProductTierPrices($productPrice, $store, $tierPrice, $tiers);
+            $this->filterTierPrices($tierPrices, $key, $tierPrice, $qtyCache, $tiers);
```

Comment:
I wonder, can we call filterTierPrices and formatProductTierPrices in different order, and functionality is not affected? 

Seems like its possible, and then the order of execution of the internal private method matches both the filterAndFormatProductTierPrices function name and its comment.

Reply:
Hi @pmarjan  I  tried call` filterTierPrices `and `formatTierPrices` in different order and got error in test.